src/main/java/net/minecraftforge/registries/ObjectHolderRegistry.java

```diff
@@ -80,24 +91,46 @@ public static void findObjectHolders()
             .map(ModFileScanData::getAnnotations)
             .flatMap(Collection::stream)
             .filter(a -> OBJECT_HOLDER.equals(a.annotationType()) || MOD.equals(a.annotationType()))
-            .collect(Collectors.toList());
+            .toList();
 
         Map<Type, String> classModIds = Maps.newHashMap();
         Map<Type, Class<?>> classCache = Maps.newHashMap();
 
-        // Gather all @Mod classes, so that @ObjectHolder's in those classes don't need to specify the mod id, Modder convince
-        annotations.stream().filter(a -> MOD.equals(a.annotationType())).forEach(data -> classModIds.put(data.clazz(), (String)data.annotationData().get("value")));
+        // Gather all @Mod classes so that @ObjectHolder's in those classes don't need to specify the mod id; modder convenience
+        annotations.stream()
+                .filter(a -> MOD.equals(a.annotationType()))
+                .forEach(data -> classModIds.put(data.clazz(), (String)data.annotationData().get("value")));
 
-        // double pass - get all the class level annotations first, then the field level annotations
-        annotations.stream().filter(a -> OBJECT_HOLDER.equals(a.annotationType())).filter(a -> a.targetType() == ElementType.TYPE)
-        .forEach(data -> scanTarget(classModIds, classCache, data.clazz(), null, (String)data.annotationData().get("value"), true, data.clazz().getClassName().startsWith("net.minecraft.")));
+        // Validate all the vanilla class-level object holders then scan those first
+        VANILLA_OBJECT_HOLDERS.forEach(data -> {
+            try
+            {
+                Class<?> holderClass = Class.forName(data.holderClass(), true, ObjectHolderRegistry.class.getClassLoader());
+                Class<?> registryClass = Class.forName(data.registryType(), true, ObjectHolderRegistry.class.getClassLoader());
```

Comment:
Why do we need to specify the class loader? Is the classloader of this class different from the thread context classloader? Add a comment explaining why.

Reply:
This is just following the convention from line 144.